src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py

```diff
@@ -103,6 +104,20 @@ def updateAgentSpeedDrainConfig(self, thresholdsHit):
 
         return
 
+    def resetAgentSpeedDrainConfig(self):
+        """
+        resetting SpeedDrainMode to False and SpeedDrainiConfig Enabled to False
+        :return:
+        """
+        if self.reqAuxDB.getWMAgentConfig(self.config.Agent.hostName).get("SpeedDrainMode"):
+            self.reqAuxDB.updateAgentConfig(self.config.Agent.hostName, "SpeedDrainMode", False)
+            speedDrainConfig = self.reqAuxDB.getWMAgentConfig(self.config.Agent.hostName).get("SpeedDrainConfig")
```

Comment:
We could use this `validKeys` thing that Erik used below just to avoid retyping the parameter name. Or we could set it as a object attr.

Reply:
In order to avoid one extra reqmgr call, how about we fetch the whole config first and then check for the keys/values?